sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -202,5 +203,9 @@ private[sql] class JacksonGenerator(
    */
   def write(array: ArrayData): Unit = writeArray(writeArrayData(array, arrElementWriter))
 
+  def write(map: MapData, mapType: MapType): Unit = {
```

Comment:
sorry to reply too late.
I thought that others `write` function don't take any type because `JacksonGenerator` has a `schema` member. Because `writeMapData` needs `mapType`, I made `write` function of `MapType` take it. 
I try to fix this problem by making `childType` matching move to `JacksonGenerator` in new commits a254f89 but I'm not sure if this way is better.
@HyukjinKwon @viirya How do you thinks about this?